sphinx/domains/python.py

```diff
@@ -91,7 +91,14 @@ def type_to_xref(text: str, env: BuildEnvironment = None) -> addnodes.pending_xr
     else:
         kwargs = {}
 
-    return pending_xref('', nodes.Text(text),
+    if env.config.python_use_unqualified_type_names:
```

Comment:
It would be better to use this option on `PyField.make_xref()`. I'll do it the next PR.